pallets/time-release/src/lib.rs

```diff
@@ -450,6 +587,122 @@ impl<T: Config> Pallet<T> {
 		Self::delete_freeze(who)?;
 		Ok(())
 	}
+
+	fn hold_balance_for(who: &T::AccountId) -> BalanceOf<T> {
+		T::Currency::balance_on_hold(&HoldReason::TimeReleaseScheduledVesting.into(), &who)
+	}
+
+	fn ensure_sufficient_hold_balance(from: &T::AccountId, amount: BalanceOf<T>) -> DispatchResult {
+		ensure!(Self::hold_balance_for(&from) >= amount, Error::<T>::InsufficientBalanceToFreeze);
+
+		Ok(())
+	}
+
+	fn validate_and_get_schedule_amount(
+		schedule: &ReleaseScheduleOf<T>,
+	) -> Result<BalanceOf<T>, DispatchError> {
+		ensure_valid_release_schedule::<T>(schedule)
+	}
+
+	fn append_release_schedule(
+		recipient: &T::AccountId,
+		schedule: ReleaseScheduleOf<T>,
+	) -> DispatchResult {
+		<ReleaseSchedules<T>>::try_append(recipient, schedule)
+			.map_err(|_| Error::<T>::MaxReleaseSchedulesExceeded)?;
+		Ok(())
+	}
+
+	fn transfer_from_free_balance(
+		from: &T::AccountId,
+		to: &T::AccountId,
+		schedule_amount: BalanceOf<T>,
+	) -> DispatchResult {
+		T::Currency::transfer(from, to, schedule_amount, Preservation::Expendable)?;
+
+		Ok(())
+	}
+
+	/// Transfers the specified amount from the hold balance of the `from` account to the `to` account.
+	///
+	/// # Arguments
+	///
+	/// * `from` - The account from which the funds will be transferred.
+	/// * `to` - The account to which the funds will be transferred.
+	/// * `schedule_amount` - The amount to be transferred.
+	///
+	/// # Errors
+	///
+	/// * `ArithmeticError::Overflow` - If the transfer amount overflows.
+	/// * `Error::InsufficientBalanceToFreeze` - If the hold balance is insufficient.
+	fn transfer_from_hold_balance(
+		from: &T::AccountId,
+		to: &T::AccountId,
+		schedule_amount: BalanceOf<T>,
+	) -> DispatchResult {
+		// Exact - Will fail if the exact amount cannot be transfered
+		// Free - deposit into the free balance but will locked following
+		// Polite - Execute with regular previlages it consideres existential
```

Comment:
I'm a little confused on these comments; perhaps a little more clarification?

Reply:
Oops, left over clean up work.

Reply:
Suggest either:
* Replace comment with a reference to https://github.com/paritytech/polkadot-sdk/blob/c8d33396345237c1864dfc0a9b2172b7dfe7ac8f/substrate/frame/support/src/traits/tokens/fungible/hold.rs#L346 for the meaning of these parameters
* Replace comment with copied text of above reference